Java/ds/CheckBST.java

```diff
@@ -0,0 +1,133 @@
+/*Solution to check if a binary tree is a Binary Search tree.
+This solution works for duplicate input values of nodes as well.
+
+Constraints:
+0<= NodeValue <= 10^4
+  
+Incorrect Approach:
+It is not enough to check if the left and right node of current node are smaller and larger than it respectively.
+This approach would't work for the following example which is actually not a Binary Search Tree --
+     15
+    /   \
+  12     28
+ /  \
+9    24 
+
+
+Correct Approach:
+It is a property of Binary Search Tree that its inorder traversal produces a sorted list in ascending order. 
+By inorder traversal of the tree we store the elements in a list and then check if the list is sorted.
+Time Complexity : O(n)
+Space Complexity : O(n)
+ */
+
+
+import java.util.*;
+
+//Class which defines the structure of each node in the binary tree
+class Node{
+	int data;
+	Node left,right;
+	Node(int data){
+		this.data = data;
+	}
+}
+
+public class CheckBST {
+
+	//Create a binary tree by taking input level wise
+	public static Node createBinaryTree() {
+		Scanner sc = new Scanner(System.in);
+		System.out.print("Enter root node value or -1 to exit: ");
+		int rootData = sc.nextInt();
+		Node root=null;
+		if(rootData!=-1) {
+			root = new Node(rootData);
+		}
+		else {
+			sc.close();
+			return root;
+		}
+		Queue<Node> tree = new LinkedList<>();
+		tree.add(root);
+		while(!tree.isEmpty()) {
+			Node temp = tree.remove();
+			System.out.print("Enter left  child of " + temp.data +" or -1 if there is no left child: ");
+			int leftChild = sc.nextInt();
+			// if leftChild is not null 
+			if (leftChild != -1) {
+				//Create leftChild Node
+				Node child = new Node(leftChild);
+				tree.add(child);
+				/* Create leftChildNode of a frontNode */
+				temp.left = child;
+			}
+			System.out.print("Enter right child of " + temp.data +" or -1 if there is no right child: ");
+			int rightChild = sc.nextInt();
+			//if rightChild is not null 
+			if (rightChild != -1) {
+				/* Create rightChild */
+				Node child = new Node(rightChild);
+				tree.add(child);
+				/* Create rightChildNode of a frontNode */
+				temp.right = child;
+			}
+		}
+		sc.close();
+		return root;
+	}
+	
+	//Correct Approach:
```

Comment:
What does `Correct Approach` is? Have you discussed an `Incorrect Approach` as well?

Reply:
Yes I have discussed the algorithmic explanation for the incorrect approach on lines 7-15. 
However there is obviously no code for it. So, I will make sure to remove this comment of Correct Approach.